File: yify/domain/src/main/java/io/github/kunal26das/yify/domain/model/Movie.kt
package io.github.kunal26das.yify.domain.model

import java.io.Serializable

@kotlinx.serialization.Serializable
data class Movie constructor(
    val id: Int,
    val backgroundImageUrl: String?,
    val coverImageUrl: String?,
    val dateUploaded: Long,
    val description: String,
    val genres: List<Genre>,
    val imdbCode: String,
    val language: Language,
    val mpaRating: String,
    val peers: Int,
    val quality: Quality,
    val rating: Float,
    val runtime: Int,
    val seeds: Int,
    val slug: String,
    val state: String,
    val summary: String,
    val synopsis: String,
    val title: String,
    val titleEnglish: String,
    val titleLong: String,
    val torrents: List<Torrent>,
    val url: String?,
    val year: Int?,
    val youtubeTrailerCode: String?,
) : Serializable {
    companion object {
        const val KEY_MOVIE = "movie"
        const val KEY_MOVIE_ID = "${KEY_MOVIE}_id"
    }
}

File: domain/entities/Movie.ts
export const POSTER_WIDTHS = [240, 480, 720] as const;

export interface Movie {
  id: number;
  imdbCode: string;
  title: string;
  titleLong: string;
  year: number;
  rating: number;
  runtimeMinutes: number;
  genres: string[];
  summary: string;
  language: string;
  mpaRating: string;
  posterUrls: string[];
  backgroundImageUrl?: string;
  ytTrailerCode?: string;
  thumbnailUrls?: string[];
}
